src/extra/utils/misc.py

```diff
@@ -30,3 +30,54 @@ def find_nearest_value(array, value: Any) -> Any:
     """
 
     return array[find_nearest_index(array, value)]
+
+
+def imshow2(image, *args, ax=None, **kwargs):
+    """Display an image with reasonable defaults.
+
+    This function wraps [plt.imshow()][matplotlib.axes.Axes.imshow] to do two
+    things automatically:
+
+    - Try to set `vmin`/`vmax` to reasonable values. Note that setting
+      `vmin`/`vmax` is incompatible with the `norm` argument, so they will only
+      be set if `norm` is not passed.
+    - Use an `auto` aspect ratio if the images aspect ratio is too skewed
+      (useful for displaying heatmaps)
+
+    All arguments other than the ones listed below are passed to
+    [plt.imshow()][matplotlib.axes.Axes.imshow], and explicitly passing any of
+    `vmin`/`vmax`/`aspect` will override the defaults.
+
+    Args:
+        image (array_like): The image to display.
+        ax (matplotlib.axes.Axes): The axis to plot the image in. This will
+            default to [plt][matplotlib.pyplot] if none is explicitly passed.
+
+    """
+    if ax is None:
+        # Try to avoid repeatedly importing matplotlib by looking in sys.modules
+        # first.
+        if "matplotlib.pyplot" in sys.modules:
+            plt = sys.modules["matplotlib.pyplot"]
+        else:
+            import matplotlib.pyplot as plt
```

Comment:
```suggestion
        import matplotlib.pyplot as plt
```

Importing something a second time is cheap - it's basically just doing the same lookup in `sys.modules` that you do here. So there's no need to complicate it. :slightly_smiling_face: 

```
In [1]: %time import matplotlib
CPU times: user 257 ms, sys: 235 ms, total: 492 ms
Wall time: 445 ms

In [2]: %time import matplotlib
CPU times: user 21 µs, sys: 0 ns, total: 21 µs
Wall time: 32.2 µs
```

Reply:
Ah cool, I thought it would re-import the whole thing. Fixed in 7906c54.